### PR TITLE
Fix fee calc problem

### PR DESCRIPTION
**Pull request by silentlight**
Ticket: https://www.notion.so/figmentnetworks/indexer-stopped-collecting-at-342401-5f3e244bb8ca44beba5892857946817e

- Use parent of parent hash for clac fee
- Bump sidecar version to 0.17.0

### CODE REVIEW

File: mappers/transaction/transaction_mappers.js [hunk 1]
```diff
@@ -28,18 +28,21 @@ const getPartialFee = (rawExtrinsic, rawEventsForExtrinsic, calcFee) => {
...
 
   const completedRawEvent =  rawEventsForExtrinsic.find(({ event }) =>
     event.section == 'system' && (event.method === 'ExtrinsicSuccess' || event.method === 'ExtrinsicFailure')
   );
 
-  if (calcFee && completedRawEvent) {
+  if (calcFee && completedRawEvent && completedRawEvent.data) {
     const completedEvent =  eventMappers.toPb(completedRawEvent);
-    const weightInfo = completedEvent.data.find(({name}) => name == 'DispatchInfo')
+    const weightInfo = completedEvent.data.find(({name}) => name == 'DispatchInfo');
```
Comment (pacmessica) [inline mappers/transaction/transaction_mappers.js:41]: maybe we should add a js linter to catch these

**Review by pacmessica — COMMENTED** [review]
why are you committing all of substrate-api-sidecar? don't we only need the vendor package we create from it?

**Comment by silentlight** [comment]
Haha! Good catch. It was a leftover from extracting tar file.

File: mappers/transaction/transaction_mappers.js (hunk 1 above)
Comment (silentlight) [reply mappers/transaction/transaction_mappers.js:41]: Yes, I am going to add a ticket for that

**Review by pacmessica — APPROVED** [review]

**Pull request merged**